engine.py

```diff
@@ -143,15 +142,36 @@ def eventFilter(self, obj, event):
 
             self._qss_watcher.fileChanged.connect(self.reload_qss)
 
+    def _add_shotgun_menu(self):
+        """
+        Add Shotgun menu to the main menu bar.
+        """
+        self.log_debug("Adding the shotgun menu to the main menu bar.")
+        self._menu_generator.create_menu()
+        self.tk_3dsmax.MaxScript.enable_menu()
+
+    def _on_menus_loaded(self, code):
+        """
+        Called when receiving CuiMenusPostLoad from 3dsMax.
+
+        :param code: Notification code received
+        """
+        self._add_shotgun_menu()
+
     def post_app_init(self):
         """
         Called when all apps have initialized
         """
         # set up menu handler
         self._menu_generator = self.tk_3dsmax.MenuGenerator(self)
-        self._menu_generator.create_menu()
+        self._add_shotgun_menu()
 
-        self.tk_3dsmax.MaxScript.enable_menu()
+        try:
+            # Listen to the CuiMenusPostLoad notification in order to add
+            # our shotgun menu after workspace reset/switch.
+            MaxPlus.NotificationManager.Register(MaxPlus.NotificationCodes.CuiMenusPostLoad, self._on_menus_loaded)
+        except AttributeError:
+            self.log_debug("CuiMenusPostLoad notification code is not available in this version of MaxPlus.")
```

Comment:
nitpick: perhaps we could also offer a further explanation, something like: The Shotgun menu will not be updated when 3dsmax workspaces are switched.